src/group/libp2p_group_gossip_server.erl

```diff
@@ -609,18 +631,33 @@ count_workers(Kind, #state{workers=Workers}) ->
     KindMap = maps:get(Kind, Workers, #{}),
     maps:size(KindMap).
 
--spec start_inbound_worker(string(), pid(), string(), #state{}) ->  #worker{}.
-start_inbound_worker(Target, StreamPid, Path, #state{tid=TID, sidejob_sup=WorkerSup}) ->
+-spec start_inbound_worker(any(), string(), pid(), string(), #state{}) ->  {noreply, #state{}}.
+start_inbound_worker(From, Target, StreamPid, Path, State = #state{tid=TID, sidejob_sup=WorkerSup, handlers=Handlers, worker_start_refs=WorkerStartRefs}) ->
     Ref = make_ref(),
+    Parent = self(),
+    spawn(fun() ->
     case sidejob_supervisor:start_child(
            WorkerSup,
            libp2p_group_worker, start_link,
            [Ref, inbound, StreamPid, self(), ?GROUP_ID, [], TID, Path]) of
         {ok, WorkerPid} ->
-            {ok, #worker{kind=inbound, pid=WorkerPid, target=Target, ref=Ref}};
+            maps:fold(fun(Key, {M, S}, Acc) ->
+                                 case (catch M:init_gossip_data(S)) of
+                                     {'EXIT', Reason} ->
+                                         lager:warning("gossip handler ~s failed to init with error ~p", [M, Reason]),
+                                         Acc;
+                                     ok ->
+                                         Acc;
+                                     {send, Msg} ->
+                                         libp2p_group_worker:send(WorkerPid, Key, Msg, true),
+                                         Acc
+                                 end
+                      end, none, Handlers),
```

Comment:
if we're not using the accumulator why not `maps:map/2` or `maps:foreach/2`?

Reply:
there's no maps:foreach in otp 22, and maps:map has sort of the same problem as fold, you're accumulating something you don't need.